site/docs/backward-compatibility.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: documentation
+title: Backward Compatibility
+---
+
+# Backward Compatibility
+
+Bazel is evolving, and we will make changes to Bazel that at times will be
+incompatible and require some changes from Bazel users.
+
+## At a glance
+
+1. Every breaking change is guarded with an `--incompatible_*` flag.
+1. Newly introduced incompatible flags default to off.
+1. For every `--incompatible_*` flag we have a github issue explaining
+   the change in behavior and giving a migration recipe.
+1. The migration window is at least one release long and is set by the author of the incompatible change.
+1. We announce what set of flags we flip with the next release one release in advance.
```

Comment:
I think the promise should be: We announce the flags we've switched in the release announcement.

It's hard to know for sure what will be switched, and we sometimes have to postpone changes. Users can know the plans by looking at GitHub labels.

Reply:
Changed the  language to "intent to flip". The point of this item is that the users can know ahead of time what's coming